### PR TITLE
Fix error when loading a model that has net_arch manually set to None

### LINKED ISSUE #1928: [Bug]: Manually setting net_arch=None causes crash when loading model
### 🐛 Bug

I've run into this issue a few times, since I use command line args in a wrapper script where I default `net_arch` to `None`, and I pass that to `policy_kwargs`.

I was able to fix it locally by changing [this line](https://github.com/DLR-RM/stable-baselines3/blob/6c00565778e5815e4589afc7499aafbd020535ae/stable_baselines3/common/base_class.py#L695) from
```python
if "net_arch" in data["policy_kwargs"] and len(data["policy_kwargs"]["net_arch"]) > 0:
```
to
```python
if data.get("policy_kwargs", {}).get("net_arch"):
```

### To Reproduce

```python
import gymnasium as gym

from stable_baselines3 import PPO

env = gym.make("CartPole-v1")

model = PPO("MlpPolicy", env, policy_kwargs=dict(net_arch=None))
model.learn(total_timesteps=5000)
model.save("ppo_cartpole")

del model

model = PPO.load("ppo_cartpole")
```

### Relevant log output / Error message

```shell
Traceback (most recent call last):
  File "/Users/hyzer/stable-baselines3/min_example.py", line 13, in <module>
    model = PPO.load("ppo_cartpole")
  File "/Users/hyzer/stable-baselines3/stable_baselines3/common/base_class.py", line 695, in load
    if "net_arch" in data["policy_kwargs"] and len(data["policy_kwargs"]["net_arch"]) > 0:
TypeError: object of type 'NoneType' has no len()
```

### System Info

- OS: macOS-14.4.1-arm64-arm-64bit Darwin Kernel Version 23.4.0: Fri Mar 15 00:19:22 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T8112
- Python: 3.10.14
- Stable-Baselines3: 2.4.0a1
- PyTorch: 2.2.2
- GPU Enabled: False
- Numpy: 1.26.4
- Cloudpickle: 3.0.0
- Gymnasium: 0.29.1

### Checklist

- [X] My issue does not relate to a custom gym environment. (Use the custom gym env template instead)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

### PR DESCRIPTION
**Pull request by jak3122**
## Description
Changed the conditional when loading the `policy_kwargs` to handle the case where the saved `net_arch` is `None`, simplified the code a bit, and added a test.

## Motivation and Context
I ran into this issue when using a wrapper script that potentially passes `None` to the `net_arch` parameter, and then when I go to load the saved model, there is an error.

closes #1928
- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/base_class.py [hunk 1]
```diff
@@ -692,10 +692,9 @@ def load(  # noqa: C901
             if "device" in data["policy_kwargs"]:
                 del data["policy_kwargs"]["device"]
             # backward compatibility, convert to new format
-            if "net_arch" in data["policy_kwargs"] and len(data["policy_kwargs"]["net_arch"]) > 0:
-                saved_net_arch = data["policy_kwargs"]["net_arch"]
-                if isinstance(saved_net_arch, list) and isinstance(saved_net_arch[0], dict):
-                    data["policy_kwargs"]["net_arch"] = saved_net_arch[0]
+            saved_net_arch = data.get("policy_kwargs", {}).get("net_arch")
```
Comment (qgallouedec) [inline stable_baselines3/common/base_class.py:695]: `"policy_kwargs"` is always in `data`, so you don't need the first `get`

**Comment by qgallouedec** [comment]
It's weird because I see you've pushed [a fix](https://github.com/DLR-RM/stable-baselines3/commit/2d62322f26048113e3a47b51af1cb732e132942d) to [my comment](https://github.com/DLR-RM/stable-baselines3/pull/1937#discussion_r1614770101) on [your branch](https://github.com/jak3122/stable-baselines3/tree/net_arch_fix), but I can't see it on the [PR page](https://github.com/DLR-RM/stable-baselines3/pull/1937). First time I've seen that.

**Comment by jak3122** [comment]
For around a day the PR page said it was processing something. Currently it says "1 workflow awaiting approval". I did forget to use the -u flag when pushing the branch originally, and only used it for the new commit, so I wonder if that broke something. I'll try pushing a dummy commit.

**Comment by jak3122** [comment]
Alright now I see the fix commit on the PR page, hope that fixed it!

**Comment by qgallouedec** [comment]
LGTM, thanks @jak3122!
You can add you id to the name of all the contributors (see bottom of changelog)

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**